src/Tools/dotnet-monitor/ConfigurationExtensions.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Configuration;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    internal static class ConfigurationExtensions
+    {
+        public static bool TryGetProviderAndValue(this IConfiguration configuration, string key, out IConfigurationProvider provider, out string value)
```

Comment:
I'm assuming some of these will go away after the merge from main?

Reply:
Yes, I rebased the changes and they're gone because they were included in #2550